query-engine/js-connectors/smoke-test-js/src/driver/neon.ts

```diff
@@ -47,23 +49,44 @@ function fieldToColumnType(fieldTypeId: number): ColumnType {
 
 type NeonConfig = PoolConfig;
 
+// return string instead of JavaScript Date object
+types.setTypeParser(1082, date => date);
+types.setTypeParser(1083, date => date);
+types.setTypeParser(1114, date => date);
+
 class PrismaNeon implements Connector, Closeable {
-  private pool: Pool
+  private pool?: Pool
+  private sql?: NeonQueryFunction<false, true>
```

Comment:
Is `Pool::query()` also an implementation of `NeonQueryFunction`? It seems to have the same function signature. You could then remove all `this.isHttp` checks by just stashing the function to use in the constructor.

Reply:
unluckily they are different things 🤣, they happened to have the same interface.